### PR TITLE
Docker image cleanup

### PR DESCRIPTION
**Pull request by daveoxley**
Add build-tools-24.0.2, do apt-get and cleanup in one command to reduce size of final image and various other clean ups.

### CODE REVIEW

**Comment by tmho** [comment]
think ill defer to @denizmveli on this one, but looks good to me :) 👍

File: Dockerfile [hunk 1]
```diff
@@ -1,33 +1,48 @@
...
 
 # Install android sdk
-RUN wget http://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
-RUN tar -xvzf android-sdk_r24.4.1-linux.tgz
-RUN mv android-sdk-linux /usr/local/android-sdk
-RUN rm android-sdk_r24.4.1-linux.tgz
+RUN curl -sf -o android-sdk_r24.4.1-linux.tgz -L http://dl.google.com/android/android-sdk_r24.4.1-linux.tgz && \
+    tar -xzf android-sdk_r24.4.1-linux.tgz && \
+    mv android-sdk-linux /usr/local/android-sdk && \
+    rm android-sdk_r24.4.1-linux.tgz
```
Comment (foodpoison) [inline Dockerfile]: i'm wondering can we make `android-sdk_r24.4.1-linux.tgz` a variable, so that we only need to update version number once

**Comment by foodpoison** [comment]
minor comment, looks good to me
👍

File: Dockerfile [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
 # Add oracle-jdk8 to repositories
-RUN add-apt-repository ppa:webupd8team/java
-
-# Update apt
-RUN apt-get update
-
 # Install oracle-jdk8
-RUN apt-get -y install oracle-java8-installer
+# Install Ruby
+RUN echo "debconf shared/accepted-oracle-license-v1-1 select true" | debconf-set-selections && \
```
Comment (denizmveli) [inline Dockerfile:8]: To me this is harder to read. What's the benefit of chaining this all into one command?

Comment (daveoxley) [reply Dockerfile:8]: Docker creates an image after every command and therefore doing the install and cleanup in one command results in less for every build to download.

**Pull request merged**